docs/modules/demos/pages/signal-processing.adoc

```diff
@@ -0,0 +1,94 @@
+** *JupyterHub*: A multi-user server for Jupyter notebooks.
+* Ingest time-series data to a timescaleDB hypertable
+* Install Jupyter notebook with a script to enrich this data with outlier detection data
+* Plot original and enriched data in pre-defined Grafana dashboards
+
+You can see the deployed products as well as their relationship in the following diagram:
+
+image::signal-processing/overview.png[]
+
+== Data ingestion
+
+The data used in this demo is a set of gas sensor measurements*. The dataset provides resistance values (r-values hereafter) for each of 14 gas sensors. In order to simulate near-real-time ingestion of this data, it is downloaded and batch-inserted into a Timescale table. It's then updated in-place retaining the same time offsets but shifting the timestamps such that the notebook code can "move through" the data using windows as if it were being streamed. The Nifi flow that does this can easily be extended to process other sources of (actually streamed) data.
+
+== JupyterHub
+
+JupyterHub will create a Pod for each active user. In order to reach the JupyterHub web interface, create a port-forward:
+
+[source,bash]
+----
+$ kubectl port-forward service/proxy-public 8000:http
+----
+
+Now access the JupyterHub web interface via:
+
+----
+http://localhost:8000
+----
+
+You should see the JupyterHub login page where you can login with username `admin` and password `adminadmin`.
+
+You should arrive at your workspace where you can click on the `notebooks` folder on the left, open the file and run it. Click on the double arrow to execute the Python scripts:
+
+image::signal-processing/notebook.png[]
+
+The notebook reads the measurement data in windowed batches using a loop, computes some predictions for each batch and persists the scores in a separate timescale table.
+
+== Model details
+
+The enriched data is calculated using an online, unsupervised https://docs.seldon.io/projects/alibi-detect/en/stable/od/methods/sr.html[model] that uses a technique called http://www.houxiaodi.com/assets/papers/cvpr07.pdf[Spectral Residuals].
+
+- `online`: in this context this means that the model does not require a training phase and can apply the algorithm to a batch in isolation. However, a light-weight training phase can be introduced if an estimated threshold value is required. This would take the form of a single batch "up front" used to determine some baselines. This is not implemented in this demo.
+- `unsupervised`: the data is not labelled and the algorithm seeks to extract useful information (alerts etc.) from the raw data alone.
+- `spectral residuals`: this algorithm executes a double Fourier transformation to reconstruct the original data, with the reconstruction error being used as a measure of outlier-ness.
+
+== Visualization
+
+Grafana can be reached by first looking up the service endpoint:
+
+[source,bash]
+----
+$ stackablectl services list
+----
+
+image::signal-processing/services.png[]
```

Comment:
We might want to use the new command `stackablectl stacklet list` and provide the output as part of the command (no screenshot).

Reply:
Yes, currently it's unavailable. However, I would like to avoid changing it again rather soon as the stackablectl `1.0.0` release is rather close. Also, some documentation already mentions the new command.

Reply:
is the output the same? If so, I'll replace the image with text.

Reply:
The output is pretty much the same (with some minor stylistic differences)

Reply:
I've changed this.

Reply:
Looks good!